test/peering-test.js

```diff
@@ -0,0 +1,249 @@
+/*!
+ * test/peering-test.js - test for peering
+ * Copyright (c) 2020, Mark Tyneway (MIT License).
+ * https://github.com/handshake-org/hsd
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const {NodeClient, WalletClient} = require('hs-client');
+const common = require('./util/common');
+const FullNode = require('../lib/node/fullnode');
+
+/**
+ * NodeFactory manages creating
+ * full nodes and the ports they
+ * listen on.
+ */
+
+class NodeFactory {
+  constructor() {
+    this.counter = 0;
+    this.PORT_COUNT = 5;
+  }
+
+  async new(config) {
+    this.counter += this.PORT_COUNT;
+    return newNode(this.counter, config);
+  }
+
+  reset() {
+    this.counter = 0;
+  }
+}
+
+const nodes = [];
+let seen = false;
+const factory = new NodeFactory();
+
+describe('Peering', function() {
+  afterEach(async () => {
+    for (const node of nodes)
+      await node.close();
+
+    nodes.length = 0;
+    factory.reset();
+    seen = false;
+  });
+
+  describe('Node only option', () => {
+    it('should create peers (plaintext)', async () => {
+      const one = await factory.new({
+        listen: true,
+        seeds: []
+      });
+
+      const addr = `127.0.0.1:${one.ports.p2p}`;
+
+      const two = await factory.new({
+        only: [addr]
+      });
+
+      nodes.push(one, two);
+
+      two.node.pool.on('peer connect', (peer) => {
+        // Peer one has a single inbound connection.
+        assert.equal(one.node.pool.peers.inbound, 1);
```

Comment:
this assertion fails intermittently on repeat testing... I wonder if you should listen for an event directly from node `one` and test it separately to avoid race condition.

Reply:
Good catch, I haven't noticed this on my machine at all. Will fix the race condition